### PR TITLE
[RayJob]: Always use target RayCluster image as default RayJob submitter image

### LINKED ISSUE #1547: [Feature] Always use RayCluster image for default RayJob submitter image
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.

### Description

Currently the RayJob submitter Job container image defaults to that of the RayCluster for compatibility reason, only when the RayCluster is created by the RayJob. In the case it was created separately, the submitter Job image defaults to `rayproject/ray:latest`, which forces to use a custom submitter template to avoid any incompatibilities.

The default RayJob submitter image should be determined from the RayCluster in which the job is meant to be submitted, in all cases.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by astefanutti**
## Related issue number

Closes #1547.

## Checks

- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by astefanutti** [comment]
@kevin85421 @z103cb @blublinsky for your review, thanks.

File: ray-operator/controllers/ray/common/job.go [hunk 1]
```diff
@@ -136,25 +136,15 @@ func GetK8sJobCommand(rayJobInstance *rayv1.RayJob) ([]string, error) {
...
 	return v1.PodTemplateSpec{
 		Spec: v1.PodSpec{
 			Containers: []v1.Container{
 				{
-					Name:  "ray-job-submitter",
-					Image: image,
+					Name: "ray-job-submitter",
+					// Use the image of the Ray head to be defensive against version mismatch issues
+					Image: rayClusterInstance.Spec.HeadGroupSpec.Template.Spec.Containers[0].Image,
 					Resources: v1.ResourceRequirements{
```
Comment (blublinsky) [inline ray-operator/controllers/ray/common/job.go:148]: Very nice
We can now remove the check in the API server

**Review by blublinsky — APPROVED** [review]
LGTM

**Review by kevin85421 — COMMENTED** [review]
Thank you for the contribution! Would you mind adding some unit tests for the function?

File: ray-operator/controllers/ray/common/job.go [hunk 2]
```diff
@@ -136,25 +136,15 @@ func GetK8sJobCommand(rayJobInstance *rayv1.RayJob) ([]string, error) {
...
+func GetDefaultSubmitterTemplate(rayClusterInstance *rayv1.RayCluster) v1.PodTemplateSpec {
 	return v1.PodTemplateSpec{
 		Spec: v1.PodSpec{
 			Containers: []v1.Container{
 				{
-					Name:  "ray-job-submitter",
-					Image: image,
+					Name: "ray-job-submitter",
+					// Use the image of the Ray head to be defensive against version mismatch issues
+					Image: rayClusterInstance.Spec.HeadGroupSpec.Template.Spec.Containers[0].Image,
```
[the remark this answers is not in the document]
Comment (astefanutti) [reply ray-operator/controllers/ray/common/job.go]: Done.

**Comment by astefanutti** [comment]
> Thank you for the contribution! Would you mind adding some unit tests for the function?

@kevin85421 I've added a unit test and re-pushed. PTAL.

**Review by kevin85421 — APPROVED** [review]
Great! Thanks!

**Pull request merged**